woocommerce/woocommerce
# [testing workflows] Add test environments and deprecate the daily e2e workflow

### Changes proposed in this Pull Request:


Partial work of https://github.com/woocommerce/woocommerce/issues/47532.

Set the ground for a test environments structure that we can build upon later.

The scope is to be able to run tests in any environment by simply running `pnpm test:e2e:with-env <environmentName>`
For this to happen I created a structure of folders for the supported environments: `envs/envName`. Each env should get its own folder containing a setup script and the playwright config.

I moved all the remaining tests from the `smoke-test-daily.yml` workflow into the new setup and deprecated the yml file, to be deleted. All the new tests projects except the Gutenberg ones were set to only run `basic.spec.js`. A proper `testMatch` or `testIgnore` pattern will be configured in follow-up PRs.

### How to test the changes in this Pull Request:

#### Existing functionality still works
Run `pnpm test:e2e-pw` - all tests in the default project should run as before.

#### Tests run for the new environments
- Run `pnpm test:e2e:with-env default` - all tests in the default project should run as before. The above command is now an alias for this one.
- `pnpm test:e2e:with-env gutenberg-stable` - all tests in the gutenberg-stable project should run
- etc.

#### Test Playwright arguments can be used
- Run `pnpm test:e2e-pw --list` - should only list the tests
- Run `pnpm test:e2e-pw smoke-tests` - should only run the smoke spec
- Run `pnpm test:e2e-pw --list shopper` - should only list the shopper tests
- etc.

#### CI is green

- CI workflow is green.
- Check the daily run (triggered [here](https://github.com/woocommerce/woocommerce/actions/runs/9350376960)) - all defined tests should run (some test failures are expected).
